apitools/base/py/credentials_lib_test.py

```diff
@@ -21,43 +25,88 @@
+        self.fail('Unexpected HTTP request to %s' % request_url)
+
+
 class CredentialsLibTest(unittest2.TestCase):
 
-    def _GetServiceCreds(self, service_account_name=None, scopes=None):
+    def _RunGceAssertionCredentials(
+        self, service_account_name=None, scopes=None, cache_filename=None):
         kwargs = {}
         if service_account_name is not None:
             kwargs['service_account_name'] = service_account_name
+        if cache_filename is not None:
+            kwargs['cache_filename'] = cache_filename
         service_account_name = service_account_name or 'default'
+        credentials = credentials_lib.GceAssertionCredentials(
+            scopes, **kwargs)
+        self.assertIsNone(credentials._refresh(None))
+        return credentials
 
-        def MockMetadataCalls(request_url):
-            default_scopes = scopes or ['scope1']
-            if request_url.endswith('scopes'):
-                return six.StringIO(''.join(default_scopes))
-            elif request_url.endswith('service-accounts'):
-                return six.StringIO(service_account_name)
-            elif request_url.endswith(
-                    '/service-accounts/%s/token' % service_account_name):
-                return six.StringIO('{"access_token": "token"}')
-            self.fail('Unexpected HTTP request to %s' % request_url)
-
-        with mock.patch.object(credentials_lib, '_GceMetadataRequest',
-                               side_effect=MockMetadataCalls,
-                               autospec=True) as opener_mock:
-            with mock.patch.object(util, 'DetectGce',
-                                   autospec=True) as mock_detect:
-                mock_detect.return_value = True
-                credentials = credentials_lib.GceAssertionCredentials(
-                    scopes, **kwargs)
-                self.assertIsNone(credentials._refresh(None))
+    def _GetServiceCreds(self, service_account_name=None, scopes=None):
+        metadatamock = MetadataMock(scopes, service_account_name)
+        with mock.patch.object(util, 'DetectGce', autospec=True) as mock_detect:
+            mock_detect.return_value = True
+            with mock.patch.object(credentials_lib, '_GceMetadataRequest',
+                    side_effect=metadatamock, autospec=True) as opener_mock:
+                credentials = self._RunGceAssertionCredentials(
+                    service_account_name=service_account_name,
+                    scopes=scopes)
             self.assertEqual(3, opener_mock.call_count)
         return credentials
 
     def testGceServiceAccounts(self):
         scopes = ['scope1']
-        self._GetServiceCreds()
-        self._GetServiceCreds(scopes=scopes)
-        self._GetServiceCreds(service_account_name='my_service_account',
-                              scopes=scopes)
+        self._GetServiceCreds(service_account_name=None,
+                                      scopes=None)
```

Comment:
Did the linter not complain about the odd spacing here (and the statement below)?

Reply:
It did, I've added a fix in the latest test. How about I squash this all

Reply:
Sounds good. That's the only nit I had -- LGTM if you're addressing that :)